docs/DeveloperGuide.md

```diff
@@ -1,38 +1,416 @@
 # Developer Guide
+## Contents
+* [Acknowledgements](#acknowledgements)
+* [Design](#design)
+    * [Architecture](#architecture)
+    * [Manager Component](#manager-component)
+    * [Profile Component](#profile-component)
+    * [TextUI Component](#textui-component)
+    * [Storage Component](#storage-component)
+    * [Parser Component](#parser-component)
+    * [Command Component](#command-component)
+* Implementation
+    * [Add a session](#add-a-session)
+    * [Remove a session](#remove-a-session)
+    * [View a session](#view-a-session)
+    * [List all sessions](#list-sessions)
+    * [Settle a session](#settle-a-session)
+    * [Add an activity](#add-an-activity)
+    * [Remove an activity](#remove-an-activity)
+    * [View an activity](#view-an-activity)
+    * [List all activities](#list-activities)
+    * [Add a group](#add-a-group)
+    * [Remove a group](#remove-a-group)
+    * [View a group](#view-a-group)
+    * [List all groups](#list-groups)
+* [Project Scope](#product-scope)
+* [User Stories](#user-stories)
+* [Non-Functional Requirements](#non-functional-requirements)
+* [Glossary](#glossary)
+
 
 ## Acknowledgements
+* Address Book (Level-3) - Provide samples of User Guide(UG) and Developer Guide(DG)
+
+## Design
+### Architecture
+![Application Diagram Screenshot](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/ArchitectureDiagram.drawio.png)
+
+The *Architecture Diagram* shown above illustrates the high-level design of the SplitLah application.
+
+**Overview of components**
+* `SplitLah [Main]`
+    * On app launch: Creates an instance of a `Manager` and runs the command loop.
+* `Manager`
+    * On creation: Initializes the Profile, TextUI and Storage components.
+    * On run: Loads data from `Storage`, receives user input from UI and uses `Parser` to parse user input into application.
+* `Profile`
+    * Handles cached data within run time of application. Manages and stores the list of `session` 
+      that were created by the user before storing it into `Storage`.
+* `TextUI`
+    * Handles UI operations of the application.
+* `Storage`
+    * Handles storage operations of the application.
+* `Parser`
+    * Handles input parsing and determines which command to run.
+* `Command`
+    * Defines how a command is executed.
+
+**Interaction between components**
+![Component Interaction Screenshot](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/ComponentInteraction.drawio.png)
```

Comment:
I think it might be confusing for the user to see 2 instances of the :SplitLah class here. You could label or explain what do each of them represent. Maybe it's possible to remove one for better comprehendability.
![image](https://user-images.githubusercontent.com/58351701/160957825-cd0378f4-92c7-4373-887e-f1f7651525d3.png)

Reply:
Overall, I think the diagram looks quite small on the website. Perhaps can consider prioritising comprehensibility over being comprehensive.